test/python/synthesis/test_permutation_synthesis.py

```diff
@@ -0,0 +1,59 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test permutation synthesis functions."""
+
+
+import unittest
+import numpy as np
+from ddt import ddt, data
+
+from qiskit.circuit.library import LinearFunction
+from qiskit.synthesis.permutation import synth_permutation_depth_lnn_kms
+from qiskit.synthesis.permutation.permutation_utils import _get_ordered_swap
+from qiskit.test import QiskitTestCase
+
+
+@ddt
+class TestPermutationSynthesis(QiskitTestCase):
+    """Test the permutation synthesis functions."""
+
+    @data(4, 5, 10, 20)
+    def test_get_ordered_swap(self, width):
+        """Test get_ordered_swap function produces correct swap list."""
+        np.random.seed(1)
+        for _ in range(5):
+            permutation = np.random.permutation(width)
+            swap_list = _get_ordered_swap(permutation)
+            output = list(range(width))
+            for i, j in swap_list:
+                output[i], output[j] = output[j], output[i]
+            self.assertTrue(np.array_equal(permutation, output))
+            self.assertLess(len(swap_list), width)
+
+    @data(4, 5, 10, 20)
+    def test_synth_permutation_depth_lnn_kms(self, width):
```

Comment:
I would suggest to check that the permutation is indeed for lnn connectivity (i.e. that all swaps are between qubits i and i+1)

Reply:
Sure. Done in [6a99690](https://github.com/Qiskit/qiskit-terra/pull/9082/commits/6a99690d81c5b9ef14d908a70e3750fcca6d0ea4).